driver/zephyr/lvgl.c

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) 2018-2019 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ *
+ * SPDX-License-Identifier: Apache-2.0
```

Comment:
@amirgon Just wanted to bring this to your attention: are you okay with having an Apache-licensed file in `lv_binding_micropython`?

Reply:
Thanks for this notice @embeddedt 

Both Apache-2.0 and MIT are permissive licenses.
Although not identical, I believe an MIT project can have Apache-2.0 sources as long as the [Apache-2.0 requirements](https://tldrlegal.com/license/apache-license-2.0-(apache-2.0)) are met:
- The copyright notice appears on the copyrighted files
- Any significant changes made to the copyrighted files must be stated
- Notice file

@svetelna - Were any changes made to the copyrighted files? Is there a NOTICE file on the original repo where you copied the copyrighted files from?

Reply:
Thank you for your response.
I did not find any NOTICE file for Zephyr.
However, I have changed the lvgl.c file a bit:
* Zephyr's LOG statements are changed to printfs
* some error checks are added
* removed ```SYS_INIT(lvgl_init, APPLICATION, CONFIG_APPLICATION_INIT_PRIO)``` macro, so that the LVGL init is not called during boot of the system.

I can state these changes in the lvgl.c.

The main reason I copied the file, and have not used its original location within Zephyr's codebase is Zephyr's Kconfig configuration. The LVGL has to be disabled in Zephyr (in order to use LVGL from the LV bindings project), thus the driver init would not work. I should have stated this in the pull request comment. I will add it there.

Reply:
> I can state these changes in the lvgl.c.

Yes that would be required from legal perspective.



> The main reason I copied the file, and have not used its original location within Zephyr's codebase is Zephyr's Kconfig configuration.

This is not clear to me.
Isn't it possible to change the original lvgl.c such that it could support LVGL with or without Micrpython?  
Duplicating code makes maintanance difficult. Future changes to the original lvgl.c would need to be merged into your copy.

Reply:
Problem with using original lvgl.c is in Kconfig defines used in the file, e.g., CONFIG_LVGL_POINTER_KSCAN.
These can not be enabled or used in Zephyr's build system, if their parent, Kconfig entry LVGL, is disabled.

I see no way how to define these within original lvgl.c. However, when the lvgl.c is copied to the Zephyr driver directory, the entries can be defined in the header file, or values directly from lv_conf.h can be used. 

If the LVGL entry in Kconfig is enabled, then the LVGL sources are compiled within Zephyr - causing redefinitions.

Reply:
> I see no way how to define these within original lvgl.c. However, when the lvgl.c is copied to the Zephyr driver directory, the entries can be defined in the header file, or values directly from lv_conf.h can be used.

There may be a way around this.
You could add to the original lvgl.c under Zephyr something like this:
```c
#if ENABLE_LV_MICROPYTHON == 1
#include "zephyr_lv_micropython.h"
#endif
```
Where `zephyr_lv_micropython.h` contains the missing definition.  
Your PR requires changes to the Makefile anyway, so it's pretty easy to add to the Makefile something like:
```makefile
CFLAGS_MOD += -DENABLE_LV_MICROPYTHON=1
```

This can prevent duplicating lvgl.c.  
What do you think?

Reply:
I think that the Zephyr itself should not have any links or references to the LV binding project, as they are not connected directly but only via the Micropython. Even Micropython is currently not mentioned anywhere in the Zephyr codebase.
It seems that from the start, the goal was to make the connection from the opposite side -> that the Micropython would contain references to Zephyr, built as the Zephyr's application.
Therefore I cannot entirely agree with the proposed solution.